### PR TITLE
Added unit test pipeline

### PR DESCRIPTION
**Pull request by M-Byte480**
Added unit test pipeline

### CODE REVIEW

File: .github/workflow/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: CI
```
Comment (Silver-Golden) [inline .github/workflow/main.yml:1]: THis is a bad name, what is this workflow for?

File: .github/workflow/unit_test.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Java CI with Maven
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:1]: This should be something like "Build"

**Review by Silver-Golden — CHANGES_REQUESTED** [review]

File: .github/workflow/unit_test.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    branches: [ "main" ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 17
```
Comment (Bourkupine) [inline .github/workflow/unit_test.yml:16]: 17 should be 21

**Review by Bourkupine — CHANGES_REQUESTED** [review]

**Review by Bourkupine — APPROVED** [review]
Looks good, is there a specific reason for using Temurin?

File: .github/workflow/main.yml
Comment (Silver-Golden) [inline .github/workflow/main.yml:1]: Honestly why is this file here?  
As far as I can tell it does nothing.  

Each file is its own workflow, you dont have to have a file to declare what other workflows exist.  
Ye cna take a look at https://forgejo.skynet.ie/Skynet/nixos/src/branch/main/.forgejo/workflows for an example of this, 3 workflows.  
You can see on the left of this page https://forgejo.skynet.ie/Skynet/nixos/actions the workflows, select each one to see the runs for that particular one.

File: .github/workflow/unit_test.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+name: Build
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:1]: This is not a build workflow, its a unit test one, update name to reflect that (or remove this line

File: .github/workflow/unit_test.yml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:8]: In the other file you have ``workflow_dispatch:``.  
This is useful to have here to allow us to manually run the workflow.

File: .github/workflow/unit_test.yml [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+jobs:
+  build:
+
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:11]: Ye can remove this line to help group the workflow together visually

File: .github/workflow/unit_test.yml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+    runs-on: ubuntu-latest
+
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:13]: Same with just above, remove this line for visual reasons

File: .github/workflow/unit_test.yml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: maven
+      - name: Build with Maven
```
Comment (Silver-Golden) [inline .github/workflow/unit_test.yml:22]: This is to run the unit test right? so ``Run unit tests using Maven`` would be more apt

**Review by Silver-Golden — CHANGES_REQUESTED** [review]
30 line PR === extensive comments  
3000 line PR === sure it looks grand

**Comment by M-Byte480** [comment]
I'll fix these tonight

**Review by Silver-Golden — APPROVED** [review]
Looks good!

**Review by Bourkupine — APPROVED** [review]
Looks good, make sure to squash commits

**Pull request merged**